src/bin/cratesfyi.rs

```diff
@@ -593,7 +606,16 @@ impl Context for BinContext {
     fn index(&self) -> Result<Arc<Index>, Error> {
         Ok(self
             .index
-            .get_or_try_init::<_, Error>(|| Ok(Arc::new(Index::new(&*self.config()?)?)))?
+            .get_or_try_init::<_, Error>(|| {
+                let config = self.config()?;
+                Ok(Arc::new(
+                    if let Some(registry_url) = config.registry_url.clone() {
```

Comment:
```suggestion
                    if config.registry_url.is_some() {
```

Reply:
The `registry_url` is used, or there's something I missing?

Reply:
Oh you're right, sorry. No need for changes.